morpheus/utils/nvt/schema_converters.py

```diff
@@ -0,0 +1,430 @@
+                     else df.assign(**{ci.name: pd.Series(None, index=df.index, dtype=ci.get_pandas_dtype())}),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[ColumnInfo] '{ci.name}'")
+        ],
+    # TODO(Devin): Custom columns are, potentially, very inefficient, because we have to run the custom function on the
+    #   entire dataset this is because NVT requires the input column be available, but CustomColumn is a generic
+    #   transform taking df->series(ci.name)
+    CustomColumn:
+        lambda ci,
+        deps: [
+            MutateOp(lambda selector,
+                     df: cudf.DataFrame({ci.name: ci.process_column_fn(df)}),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[CustomColumn] '{ci.name}'")
+        ],
+    DateTimeColumn:
+        lambda ci,
+        deps: [
+            Rename(f=lambda name: ci.name if name == ci.input_name else name),
+            LambdaOp(lambda series: series.astype(ci.dtype), dtype=ci.dtype, label=f"[DateTimeColumn] '{ci.name}'")
+        ],
+    IncrementColumn:
+        lambda ci,
+        deps: [
+            MutateOp(partial(
+                nvt_increment_column, output_column=ci.groupby_column, input_column=ci.name, period=ci.period),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.groupby_column)],
+                     label=f"[IncrementColumn] '{ci.name}' => '{ci.groupby_column}'")
+        ],
+    RenameColumn:
+        lambda ci,
+        deps: [
+            MutateOp(lambda selector,
+                     df: df.rename(columns={ci.input_name: ci.name}),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[RenameColumn] '{ci.input_name}' => '{ci.name}'")
+        ],
+    StringCatColumn:
+        lambda ci,
+        deps: [
+            MutateOp(partial(nvt_string_cat_col, output_column=ci.name, input_columns=ci.input_columns, sep=ci.sep),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[StringCatColumn] '{','.join(ci.input_columns)}' => '{ci.name}'")
+        ],
+    StringJoinColumn:
+        lambda ci,
+        deps: [
+            MutateOp(partial(
+                nvt_string_cat_col, output_column=ci.name, input_columns=[ci.name, ci.input_name], sep=ci.sep),
+                     dependencies=deps,
+                     output_columns=[(ci.name, ci.dtype)],
+                     label=f"[StringJoinColumn] '{ci.input_name}' => '{ci.name}'")
+        ],
+    JSONFlattenInfo:
+        lambda ci,
+        deps: [json_flatten_from_input_schema(ci.input_col_names, ci.output_col_names)]
+}
+
+
+def build_nx_dependency_graph(column_info_objects: typing.List[ColumnInfo]) -> nx.DiGraph:
```

Comment:
If this is only used in this file and in tests, prefix with an underscore to show users shouldnt use this function.